### PR TITLE
New about page

### LINKED ISSUE #148: Responsive Layout for testimonials / About page
Follow up from previous comments: https://github.com/ourjapanlife/findadoc-frontend/pull/79#pullrequestreview-687824304

and https://github.com/ourjapanlife/findadoc-frontend/pull/79#issuecomment-864427326

Check with Teresa on this as she is working on the About Page layout

### PR DESCRIPTION
**Pull request by renderf0x**
## Resolves #148 

Creates a new about page layout and widgets to follow @thehamspecial 's [design mocks](https://www.figma.com/file/kdgcV9gd3cOeiSgoXY6VLM/Find-a-Doc?node-id=0%3A1).

As part of this, moves the Twitter API calls to a service package to separate the logic.

** Out of scope **
The following items need to be addressed, but I believe they should be in separate PRs:
- Absolute text should be replaced with String IDs for localization
- String IDs and text should be moved into `data/` and loaded

I want to separate this page from that effort, so will save that for a followup (esp since we don't have the strings defined yet).

### How to test

- Vuetify preview - check against mocks
- Locally - check against mocks

### Screenshots

スクリーンショット 2021-07-23 16 36 24

スクリーンショット 2021-07-23 16 36 33

スクリーンショット 2021-07-23 16 36 49

スクリーンショット 2021-07-23 16 37 04

スクリーンショット 2021-07-23 16 37 13

### CODE REVIEW

**Comment by renderf0x** [comment]
@thehamspecial please take a look at this when you get a chance (the netlify preview)

There's currently a first-load issue again on the tweets, so you'll need to do one refresh, but the layout should be there regardless. Let me know if you want changes.

**Comment by renderf0x** [comment]
> There's currently a first-load issue again on the tweets, so you'll need to do one refresh, but the layout should be there regardless. Let me know if you want changes.

Fixed via `asycnData`.

**Review by theyokohamalife — COMMENTED** [review]
It’s looking great so far! I noticed a small issue while swiping on my phone to see the tweets. The page extends further right than intended. Maybe this is due to the large image at the top?

C3B05551-EE36-4EE1-95C1-2AA811541EA7
320EB393-2D80-497E-9820-09E22AA47F3F

File: components/about/KofiWidget.vue [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<template>
```
Comment (ermish) [inline components/about/KofiWidget.vue:1]: ty for grouping the components! we've finally been creating too many for a flat tree

File: components/about/Header.vue [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<template>
+  <div>
+    <div class="header-widget">
```
Comment (ermish) [inline components/about/Header.vue:3]: beautiful

File: components/about/KofiWidget.vue [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                Donate a cuppa
+              </v-btn>
+            </v-row>
+          </div>
+        </v-col>
+      </v-row>
+    </v-container>
+  </div>
+</template>
+<style>
```
Comment (ermish) [inline components/about/KofiWidget.vue:27]: very clean component!

File: components/about/KofiWidget.vue [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  <div class="kofi-widget">
+    <v-container>
+      <v-row>
+        <v-col sm="12" class="kofi-widget-content-wrap">
+          <div class="kofi-widget-content">
+            <v-row>
+              <h2>Buy us coffee ☕</h2>
+            </v-row>
+            <v-row>
+              <p>We are running this database after our 9-5 jobs. Help us stay awake!</p>
```
Comment (ermish) [inline components/about/KofiWidget.vue:11]: you said you're moving the translations in a separate PR correct?

File: components/about/SiteTimeline.vue [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      <v-timeline-item
+        v-for="item in items"
+        :key="item.date"
+        fill-dot
+        small
+        color="#96DCE8"
+      >
+      <v-container>
+        <v-row>
+            <v-col :cols="$vuetify.breakpoint.mobile ? 12 : 4">
```
Comment (ermish) [inline components/about/SiteTimeline.vue:12]: should we accounting for a middle tier on this? ipads and such?

File: components/about/SiteTimeline.vue [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+              <h2>
+                {{ item.date }}
+              </h2>
+              <div class="timeline-date-headline">
+                {{ item.dateHeadline }}
+              </div>
+            </v-col>
+            <v-col :cols="$vuetify.breakpoint.mobile ? 12 : 7">
```
Comment (ermish) [inline components/about/SiteTimeline.vue:20]: hmmm, I wonder if it's possible to make cols dynamic/responsive based on screen size instead of hard breakpoints. 
I haven't checked but does v-col support this instead of a fixed passed in param?

File: components/about/Team.vue [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    <AboutSvgWavyBump />
+  </div>
+</template>
+<script>
+import { leads } from "../../data/contributors.json";
+
+export default {
+  data() {
+    return {
+      leads: leads
```
Comment (ermish) [inline components/about/Team.vue:49]: should avoid using identical variable names. 
maybe call the left assignment, `teamMembers` to match the component context.

File: components/about/Team.vue [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+  .about-team {
+    color:  white;
+  }
+
+  .about-team a {
+    color:  white;
+  }
+
+  .about-team h2 {
+    font-family: Montserrat;
```
Comment (ermish) [inline components/about/Team.vue:64]: We should follow consistent font types throughout the app. Is this meant to be our secondary font type or a one off?

File: components/about/TeamMember.vue [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    },
+    twitterHandle: String,
+    customPic: String,
+  },
+  computed: {
+    githubAvatarUrl: function () {
+      return `${getGithubProfileUrl(this.githubHandle)}.png`;
+    },
+    githubProfileUrl: function () {
+      return getGithubProfileUrl(this.githubHandle);
```
Comment (ermish) [inline components/about/TeamMember.vue:65]: nit: not sure it's worth making a method for this `get` method for only two spots and then not for twitter version. 😄 
NBD, just pointing it out

File: components/about/Trust.vue [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+                </div>
+            </v-col>
+        </v-row>
+      <v-row>
+        <v-col
+            v-for="card in cards"
+            :key="card.avatar"
+        >
+          <v-card width="360px" height="397px" class="my-auto">
+            <v-card-title class="justify-center align-center">
```
Comment (ermish) [inline components/about/Trust.vue:18]: can you move these classes to css?

File: components/about/Tweet.vue [hunk 11]
```diff
@@ -43,20 +43,39 @@
 export default {
   props: {
     tweet: {
-      required: true,
       type: Object,
+      required: true
     },
+    turquoise: {
```
Comment (ermish) [inline components/about/Tweet.vue:49]: instead of just `turqoise` should make it relevant to "why" it's turqoise.  If you really want turqoise, then should at least add a verb since it's a boolean. `isTurqoise` .

File: components/about/Tweet.vue [hunk 12]
```diff
@@ -43,20 +43,39 @@
...
+      type: Boolean,
+      required: false
+    }
   },
+  methods: {
+    avatarColorClasses(turquoise) {
```
Comment (ermish) [inline components/about/Tweet.vue:55]: it might be easier to make conditional components/elements instead of manipulating so many classes. Would probably be easier for the next contributor to read

File: components/about/TwitterTestimonials.vue [hunk 13]
```diff
@@ -0,0 +1,95 @@
+<template>
+  <div
+    class="mx-auto tweet-widget"
```
Comment (ermish) [inline components/about/TwitterTestimonials.vue:3]: nit: `mx-auto` could be moved to css

File: components/about/TwitterTestimonials.vue [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+  </div>
+</template>
+<script>
+export default {
+  props: {
+    tweets: {
+      type: Array,
+      required: true
+    },
+    turquoise: {
```
Comment (ermish) [inline components/about/TwitterTestimonials.vue:35]: same as other turqoise comment.

File: pages/about.vue [hunk 15]
```diff
@@ -1,39 +1,55 @@
...
+<script>
+import TwitterAPI from "../services/twitter";
 
+export default {
+  data() {
+    return {
+      tweets: [],
+    };
+  },
+  async asyncData({ $axios }) {
```
Comment (ermish) [inline pages/about.vue:43]: would `getTweets` be a better name?

File: pages/about.vue [hunk 16]
```diff
@@ -1,39 +1,55 @@
...
+    const twitter = new TwitterAPI($axios);
+    const tweets = await twitter.get();
```
Comment (ermish) [inline pages/about.vue:45]: should this be `await`ed?

File: services/twitter.js [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+        ].profile_image_url.replace(/_normal\./, "."),
+      });
+      annotated.push(annotatedTweet);
+    }
+    return annotated;
+  }
+
+  async get() {
+    const tweets = await this._fetch();
+    const userMap = this._createUserMap(tweets.includes.users);
```
Comment (ermish) [inline services/twitter.js:77]: should validate the `tweets` object since it's coming from an external source.

File: services/twitter.js [hunk 18]
```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-console */
```
Comment (ermish) [inline services/twitter.js:1]: thoughts on suffixing as a service to make it more apparent? `twitterService`. 
Smurf naming is always a fun debate. 

tbh, when I was reading `new Twitter` earlier, I wasn't sure what the class did. It would have been nice to know without looking it up

**Review by ermish — CHANGES_REQUESTED** [review]
Looks really great! See lots of good practices in here and visually looks awesome! I added some small comments. @renderf0x

File: services/twitter.js (hunk 18 above)
Comment (renderf0x) [reply services/twitter.js:1]: I think that's a good name for the class. I dunno if we need to call the file that since it's already in `services/`.
I'll change the class name to TwitterService.

Comment (ermish) [reply services/twitter.js:1]: brilliant!

File: pages/about.vue (hunk 16 above)
Comment (renderf0x) [reply pages/about.vue:45]: yep.

File: pages/about.vue (hunk 15 above)
Comment (renderf0x) [reply pages/about.vue:43]: yeah I debated over that. `getTweets` is fine, can change it from `get` to that.

File: components/about/Tweet.vue (hunk 12 above)
Comment (renderf0x) [reply components/about/Tweet.vue:55]: I think you're right here; I think I was approaching it from a 'what if we have multiple conditions' standpoint, but we only have the one, so we can just ternary here.

Comment (renderf0x) [reply components/about/Tweet.vue:55]: Ok, I take that back. I think we need to do it this way to keep the other colors on there.

The above comment was made on the assumption that static and interpolated values would be combined, like with css classes in Vue. That would look something like this:
`color="lighten-2 avatar-border" :color="turquoise ? 'white' : 'cyan'"`

However, that doesn't work, likely because of the way that Vuetify implements the avatar component.

The other way of course could be swapping out whole elements, but I think that's way more confusing as it's a bunch of duplication for no apparent purpose.

File: components/about/Tweet.vue (hunk 11 above)
Comment (renderf0x) [reply components/about/Tweet.vue:49]: I was attempting to follow the Vuetify style color naming here. I'm not 100% set on it tho.

File: components/about/Team.vue (hunk 8 above)
Comment (renderf0x) [reply components/about/Team.vue:64]: This is meant to match Teresa's design, in which the large headers are often, but not always, Monserrat. I agree that it'd be great to migrate all these up to a global when we get SCSS in.

File: components/about/Team.vue (hunk 7 above)
Comment (renderf0x) [reply components/about/Team.vue:49]: Sure? Since the context is different there's no collision, but I can change.

File: components/about/SiteTimeline.vue (hunk 6 above)
Comment (renderf0x) [reply components/about/SiteTimeline.vue:20]: There are ways to do that, but my interpretation of the design mocks was that mobile was separate from desktop layout, hence all the breaking. Is this not the case?

File: components/about/SiteTimeline.vue (hunk 5 above)
Comment (renderf0x) [reply components/about/SiteTimeline.vue:12]: Perhaps, but we don't currently have a mock for a middle tier, so that's out of scope. Though, one could surely be made.

File: components/about/KofiWidget.vue (hunk 4 above)
Comment (renderf0x) [reply components/about/KofiWidget.vue:11]: Yeah, my plan was to merge this, then spin out the string ids after. My logic there is - we're unlikely to have translations for any of this off the bat anyways, until someone translates it, so a fast-follow should be fine, like we're doing with News.

File: components/about/KofiWidget.vue (hunk 3 above)
Comment (renderf0x) [reply components/about/KofiWidget.vue:27]: ty!

File: components/about/KofiWidget.vue (hunk 1 above)
Comment (renderf0x) [reply components/about/KofiWidget.vue:1]: Yeah it turns out that the Vue / Nuxt system for this is fairly easy to reason with!

File: components/about/TeamMember.vue (hunk 9 above)
Comment (renderf0x) [reply components/about/TeamMember.vue:65]: > It’s looking great so far! I noticed a small issue while swiping on my phone to see the tweets. The page extends further right than intended. Maybe this is due to the large image at the top?
> 

Thanks @theyokohamalife !

So this issue is caused by page overflow, and I'd noticed it before (it happens on our current page too) but had forgotten about it. If we disable horizontal scroll/overflow on the root element for this it will fix.

File: services/twitter.js (hunk 17 above)
Comment (renderf0x) [reply services/twitter.js:77]: What do we think the behavior should be on a failed fetch?

Should we
a) Load fixtures, since this isn't mission critical
or
b) Render empty widget
or
c) Fail render
?

File: components/about/Trust.vue (hunk 10 above)
Comment (renderf0x) [reply components/about/Trust.vue:18]: w/h or content centering? I actually think the latter are rather helpful, but there's no reason w/h need to be inline.

File: components/about/SiteTimeline.vue (hunk 5 above)
Comment (ermish) [reply components/about/SiteTimeline.vue:12]: Sounds like a good follow up design for later 👌

File: components/about/SiteTimeline.vue (hunk 6 above)
Comment (ermish) [reply components/about/SiteTimeline.vue:20]: I think that's correct for some feature breakpoints, but we should also provide some responsive support within these breakpoints to utilize different screen sizes. 
for example, 
- desktop layout-> 4columns (old ipad resolution) to 12 columns (deskop ultrawide resolution). 
- mobile layout->  responsive 100% width until breakpoint. 

cc @thehamspecial

File: services/twitter.js (hunk 17 above)
Comment (ermish) [reply services/twitter.js:77]: hmmmmmmm...fixtures make sense to me since it will still provide a nice experience

File: components/about/Team.vue (hunk 8 above)
Comment (theyokohamalife) [reply components/about/Team.vue:64]: I set Monserrat as the default font for headings across the site in `assets/variable.scss` in a [previous PR](https://github.com/ourjapanlife/findadoc-frontend/pull/248). It should work for you if you pulled the latest changes from Main.

File: components/about/Header.vue [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+      <div class="header-widget-content">
+        <h1>About Find A Doc, the database reducing vaccine waste.</h1>
```
Comment (ann-kilzer) [inline components/about/Header.vue:5]: Great! Can we add i18n values in, too?

File: components/about/KofiWidget.vue (hunk 4 above)
Comment (ann-kilzer) [reply components/about/KofiWidget.vue:11]: Ah fast-follow is good 👍

File: services/twitter.js (hunk 17 above)
Comment (renderf0x) [reply services/twitter.js:89]: Done. I un-prefixed the 'TEST' I had in the beginning of the fixture strings now that there's a chance we might use them on the prod page. I'd thought about this before too; just hadn't done it.

File: components/about/Header.vue (hunk 19 above)
Comment (renderf0x) [reply components/about/Header.vue:5]: Fast follow after =)

File: components/about/TeamMember.vue (hunk 9 above)
Comment (renderf0x) [reply components/about/TeamMember.vue:65]: @theyokohamalife fixed by adding global overflow-x limitations.

File: components/about/Team.vue (hunk 8 above)
Comment (renderf0x) [reply components/about/Team.vue:64]: @theyokohamalife Ah yeah, I see them. These are set only to the Vue header classes (ie `.text-h3`) and not the the actual elements. I think we should follow-on to that and set the element defaults as well, but also that should be done separately and examined globally.

I think we should leave this for now, until we do the above - I'd rather not just replace one extra line we'll need to delete later with another, as `<h3 class="text-h3"` is redundant.

File: components/about/TwitterTestimonials.vue (hunk 13 above)
Comment (renderf0x) [reply components/about/TwitterTestimonials.vue:3]: Could be, however, I'd argue that one of the things that using Vuetify gives us is convenience classes...

File: components/about/TwitterTestimonials.vue (hunk 14 above)
Comment (renderf0x) [reply components/about/TwitterTestimonials.vue:35]: Again, I think leaving this as is is actually more in line with the way Vuetify handles these.

**Comment by renderf0x** [comment]
@ermish I think I've gone through the comments - let me know if you have anything you insist on that's left.

File: components/about/Trust.vue (hunk 10 above)
Comment (ermish) [reply components/about/Trust.vue:18]: I meant the classes `justify-center` `align-center`, but no big deal since I'm reworking our css classes.

File: components/about/Tweet.vue (hunk 11 above)
Comment (ermish) [reply components/about/Tweet.vue:50]: what naming do they have for prop names?

Comment (ermish) [reply components/about/Tweet.vue:50]: it seems really strange that vuetify suggests naming booleans without verbs in them. That's typically a standard practice....I don't think we should always use vuetify as a truth for naming conventions and best practices 🤔

File: pages/about.vue (hunk 15 above)
Comment (ermish) [reply pages/about.vue:43]: I actuall meant instead of `asyncData` 😅

File: services/twitter.js (hunk 17 above)
Comment (ermish) [reply services/twitter.js:89]: we also have a `data` folder we should move it to instead of testfixtures, since it's prod data.  (at least if we plan to keep two separate folders for static data)

**Review by ermish — APPROVED** [review]
Changes look good enough if you want to merge in and potentially clean up anything after. At this point, I just want us to get this merged in!

File: pages/about.vue (hunk 15 above)
Comment (renderf0x) [reply pages/about.vue:43]: Ah, asyncData is actually a [nuxt hook](https://nuxtjs.org/docs/2.x/features/data-fetching) that fetches data for the entire page. It's a 'magic name'.

Comment (renderf0x) [reply pages/about.vue:43]: We could theoretically refactor this to have a different getTweets function that gets referenced here, but with the logic so straightforward it doesn't make sense yet I think.

File: services/twitter.js (hunk 17 above)
Comment (renderf0x) [reply services/twitter.js:89]: Yeah, was debating on that. I could go either way on it, as it really is intended to only be fallback and we don't want to give the impression that this is where we canonically want to load from, but otoh, we now have the possibility of using this if things go south.

**Comment by renderf0x** [comment]
> Changes look good enough if you want to merge in and potentially clean up anything after. At this point, I just want us to get this merged in!

Awesome, done. I can then start getting the i18n strings ported within the next day or two.

File: pages/about.vue (hunk 15 above)
Comment (ermish) [reply pages/about.vue:43]: ahhhh fair enough. another magic name 🔥

**Comment by renderf0x** [comment]
Fast follow for keys:

https://github.com/ourjapanlife/findadoc-localization/pull/110

I can put ja translations in after we merge this.
